e2e/_suites/ingest-manager/features/agent_endpoint_integration.feature

```diff
@@ -0,0 +1,38 @@
+@agent_endpoint_integration
+Feature: Fleet Mode Agent with Endpoint Integration
+  Scenarios for Agent to deploy Endpoint and sending data to Ingest Manager and Elasticsearch.
+
+@deploy-endpoint-with-agent
+Scenario: Deploy Agent and add Endpoint Integration and host shows in Security App
+  Given an agent is deployed to Fleet
+  When the "Endpoint" latest package version is installed successfully
+    And the "Endpoint" integration is "added" in the "default" configuration
+  Then the "default" configuration shows the "Endpoint" datasource added
+    And the host name is shown in the security app
+    
+@endpoint-policy-check
+Scenario: Deploy Endpoint and see policy return successful in Security App
+  Given an Endpoint is successfully deployed with Agent
+  Then the policy response will be listed in the security app
+
+@set-policy-and-check-changes
+Scenario: Deploy Endpoint and change default policy and verify in Security App
+  Given an Endpoint is successfully deployed with Agent
+  When the policy is updated to have malware in detect mode
+  Then the policy response will be listed in the security app
+
+@deploy-endpoint-then-unenroll-agent-and-verify
+Scenario: Deploy Endpoint and then un-enroll Agent to confirm Endpoint stops
+  Given an Endpoint is successfully deployed with Agent
+  When the agent is un-enrolled
+  Then the agent is not listed in Fleet as online
+    And the endpoint is not listed in Security App as online
+    And the "elastic-endpoint" process is "stopped" on the host
+
+@deploy-endpoint-then-remove-it-from-configuration-and-verify
+Scenario: Deploy Endpoint and remove it from the configuration then confirm Endpoint stops
+  Given an Endpoint is successfully deployed with Agent
+  When the "Endpoint" integration is "removed" in the "default" configuration
+  Then the agent is listed in Fleet as online
+    And the endpoint is not listed in Security App as online
```

Comment:
Will this happen to any enrolled Endpoint? Said in other words: should we test with multiple endpoints connected?

Reply:
Indeed any Agent running with that Agent Configuration will be updated.  If we want to update the language now to be future complaint with more advanced tests that would be cool.

It would be great to spin up a 2nd Docker container and to test the multiple Agents case.  Definitely don't want to get into that just yet - but if we had a test that exemplified how to spin up multiple Docker Agents it would be great - I'll put a ticket for this.  I also, think testing multiple Agents as such is moderate to low risk, so not a priority.  True scale testing is a higher risk and priority, but that would need to be in the hundreds or even thousands to be relevant.  

Having said that... I don't think any other test frame-work is well suited to testing multiple Agents at the same time - so this may be a great chance to fill in some coverage not otherwise supported.